vault/identity_store_oidc_provider.go

```diff
@@ -1162,6 +1171,11 @@ func (i *IdentityStore) pathOIDCCreateUpdateProvider(ctx context.Context, req *l
 	var resp logical.Response
 	name := d.Get("name").(string)
 
+	if name == defaultProviderName {
```

Comment:
We might be able to allow modification of the default provider. For example, to add some scopes or change the issuer.

Reply:
I'm thinking we should allow the default provider and key to be modified but not deleted. I don't see why not, and I'm sure there will be a request to do so.

Reply:
I agree. I say allow modification for default provider and key.

Reply:
Done in b55d79b.